tests/test-theme-fonts.php

```diff
@@ -214,6 +264,117 @@ public function test_get_all_fonts_from_theme_and_variation() {
 
 	}
 
+	public function test_non_array_font_src() {
+		wp_set_current_user( self::$admin_id );
+
+		$test_theme_slug = $this->create_blank_theme();
+
+		// Create a theme with a non-array font src
+		$theme_json = CBT_Theme_JSON_Resolver::get_theme_file_contents();
+		$theme_json['settings']['typography']['fontFamilies'] = array(
+			array(
+				'slug'       => 'single-src-font',
+				'name'       => 'Single Src Font',
+				'fontFamily' => 'Single Src Font',
+				'fontFace'   => array(
+					array(
+						'fontFamily' => 'Single Src Font',
+						'fontStyle'  => 'normal',
+						'fontWeight' => '400',
+						'src'        => 'file:./assets/fonts/single-src-font.ttf',
+					),
+				),
+			),
+		);
+		CBT_Theme_JSON_Resolver::write_theme_file_contents( $theme_json );
+
+		// Attempt to get all fonts
+		$fonts = CBT_Theme_Fonts::get_all_fonts();
+
+		// Verify that the src is correctly handled
+		$this->assertCount( 1, $fonts );
+		$this->assertEquals( 'single-src-font', $fonts[0]['slug'] );
+		$this->assertEquals( get_stylesheet_directory_uri() . '/assets/fonts/single-src-font.ttf', $fonts[0]['fontFace'][0]['src'] );
+
+		$this->uninstall_theme( $test_theme_slug );
+	}
+
+	public function test_array_font_src() {
+		wp_set_current_user( self::$admin_id );
```

Comment:
This is probably not needed.

Reply:
Without it, the tests break:

```php
Time: 00:00.357, Memory: 42.50 MB

There was 1 error:

1) Test_Create_Block_Theme_Fonts::test_copy_activated_fonts_to_theme
Undefined array key "typography"

/var/www/html/wp-content/plugins/create-block-theme/tests/test-theme-fonts.php:43

ERRORS!
Tests: 67, Assertions: 163, Errors: 1.
Script phpunit handling the test event returned with error code 2
✖ Command failed with exit code 2
Command failed with exit code 2
```

Reply:
> Probably we should use this refactor to improve the tests of this class. They currently relay on setting users, creating new themes dynamically and after removing it without restoring the previous state.

I agree that we should work on that, but since the tests currently cover the already-refactored code, I'd prefer refactoring them in another PR.